zivid_camera/src/zivid_camera.cpp

```diff
@@ -449,40 +419,59 @@ bool ZividCamera::isConnectedServiceHandler(IsConnected::Request&, IsConnected::
 
 void ZividCamera::publishFrame(Zivid::Frame&& frame)
 {
-  const bool publish_points = shouldPublishPoints();
+  const bool publish_points_xyz = shouldPublishPointsXYZ();
+  const bool publish_points_xyzrgba = shouldPublishPointsXYZRGBA();
   const bool publish_color_img = shouldPublishColorImg();
   const bool publish_depth_img = shouldPublishDepthImg();
+  const bool publish_snr_img = shouldPublishSnrImg();
 
-  if (publish_points || publish_color_img || publish_depth_img)
+  if (publish_points_xyz || publish_points_xyzrgba || publish_color_img || publish_depth_img || publish_snr_img)
   {
     const auto header = makeHeader();
-    const auto point_cloud = frame.getPointCloud();
+    auto point_cloud = frame.pointCloud();
+
+    // Transform point cloud from millimeters (Zivid SDK) to meter (ROS).
+    const float scale = 0.001f;
+    const auto transformationMatrix = Zivid::Matrix4x4{ scale, 0, 0, 0, 0, scale, 0, 0, 0, 0, scale, 0, 0, 0, 0, 1 };
+    point_cloud.transform(transformationMatrix);
 
-    if (publish_points)
+    if (publish_points_xyz)
     {
-      publishPoints(header, point_cloud);
+      publishPointCloudXYZ(header, point_cloud);
     }
-
-    if (publish_color_img || publish_depth_img)
+    if (publish_points_xyzrgba)
     {
-      const auto camera_info = makeCameraInfo(header, point_cloud.width(), point_cloud.height(), camera_.intrinsics());
+      publishPointCloudXYZRGBA(header, point_cloud);
+    }
+    if (publish_color_img || publish_depth_img || publish_snr_img)
+    {
+      const auto intrinsics = Zivid::Experimental::Calibration::intrinsics(camera_);
```

Comment:
Could the be any downsides to using experimental features in the wrapper?

Reply:
This was moved into experimental in 2.x. We still need it in ROS. This probably means we cannot remove this function from Zivid::Experimental until we bump the major version again.